docs/cudf/source/user_guide/copy-on-write.md

```diff
@@ -0,0 +1,99 @@
+# Copy on write
+
+Copy on write enables ability to save on GPU memory usage when deep copies of a column
+are made.
+
+## How to enable it
+
+i. Either by using `set_option` in `cudf`:
+
+```python
+>>> import cudf
+>>> cudf.set_option("copy_on_write", True)
+```
+
+ii. Or, by setting an environment variable ``CUDF_COPY_ON_WRITE`` to ``1`` prior to the
+launch of the python interpreter:
+
+```bash
+export CUDF_COPY_ON_WRITE="1"
+```
+
+
+## Making copies
+
+There are no additional changes required in the code to make use of copy-on-write.
+
+```python
+>>> series = cudf.Series([1, 2, 3, 4])
+```
+
+Performing a deep copy will create a new series object but pointing to the
+same underlying device memory:
+
+```python
+>>> copied_series = series.copy(deep=True)
+>>> series
+0    1
+1    2
+2    3
+3    4
+dtype: int64
+>>> copied_series
+0    1
+1    2
+2    3
+3    4
+dtype: int64
+>>> 
+>>> 
+>>> series.data.ptr
+140102175031296
+>>> copied_series.data.ptr
+140102175031296
+```
```

Comment:
Is this still the case based on the latest changes? If so I would argue this is still unsafe and unexpected behavior because we don't know what someone will be doing with `ptr` and we're potentially violating their ask for a deep copy.

Reply:
Question from my side. The docs say

> Performing a deep copy will create a new series object but pointing to the
same underlying

If a deep copy creates a new object but pointing to the same data, whet does a shallow copy do then? (`copy(deep=False`) Or is there just no difference in cudf?

Reply:
Yes, there is no difference with a shallow copy(when c-o-w is enabled). However, looking at this comment:

> I responded about this in another inline comment at https://github.com/rapidsai/cudf/pull/11718#discussion_r1008231500, but so currently copy(deep=True) in pandas gives an actual physical copy that doesn't share memory (or track references) with the parent.
(while deep=False does the CoW protected view)

We probably need to toggle and align cudf's copy-on-write behavior